test/join.jl

```diff
@@ -112,9 +112,62 @@ module TestJoin
 
     # Test that join works when mixing Array and NullableArray (#1151)
     dt = DataTable([collect(1:10), collect(2:11)], [:x, :y])
-    dtnull = DataTable(x = 1:10, z = 3:12)
+    dtnull = DataTable(x = NullableArray(1:10), z = NullableArray(3:12))
```

Comment:
It'll be useful to make this explicit now so the test will still work when columns are not NullableArrays by default